sdk/go/common/resource/property_compatibility_test.go

```diff
@@ -42,3 +45,37 @@ func TestRoundTripConvert(t *testing.T) {
 		assert.True(t, source.Equals(roundTripped))
 	})
 }
+
+func TestConversionThroughGRPC(t *testing.T) {
+	t.Parallel()
+
+	m := property.New(map[string]property.Value{
+		"known":   property.New("v1"),
+		"output":  property.New(property.Computed).WithSecret(true),
+		"unknown": property.New(property.Computed),
+	})
+
+	rm := resource.ToResourcePropertyValue(m)
+
+	mm, err := plugin.MarshalPropertyValue("", rm, plugin.MarshalOptions{
+		KeepUnknowns:     true,
+		KeepOutputValues: true,
+	})
+	require.NoError(t, err)
+
+	nrm, err := plugin.UnmarshalPropertyValue("", mm, plugin.MarshalOptions{
+		KeepUnknowns:     true,
+		KeepOutputValues: true,
+	})
+	require.NoError(t, err)
+
+	// Inexplicably, the resource.PropertyMap survive do not survive round
+	// tripping. We see a the empty string in rm turn into a nil value in nrm. This is
+	// semantically equivalent since they are both in a resource.Computed and thus the
+	// value doesn't matter, but it's pretty painful.
+	assert.NotEqual(t, rm, nrm)
```

Comment:
I don't think we care about this assert. If we changed the Unmarshal/Marshal code to fix this we wouldn't want this test to break. Worth keeping the comment though.

Reply:
I put it there so we would know to remove the comment when if we changed Unmarshal/Marshal. I can remove it.